modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java

```diff
@@ -0,0 +1,593 @@
+                || ApiBrowserConstants.check(url, ApiBrowserConstants.VIEW_OPERATION)) {
+            point = ApiBrowserConstants.LIST_OPERATIONS;
+        } else if (ApiBrowserConstants.check(url, ApiBrowserConstants.VIEW_DOCUMENTATION)) {
+            point = ApiBrowserConstants.VIEW_DOCUMENTATION;
+        }
+        if (point == null) {
+            // check plugins
+            List<Plugin<?>> plugins = getSnapshotManager().getPlugins();
+            for (Plugin<?> plugin : plugins) {
+                point = plugin.getView(url);
+                if (point != null) {
+                    break;
+                }
+            }
+        }
+
+        return point;
+    }
+
+    @GET
+    @Produces("text/html")
+    public Object doGet() {
+        return getView("index").arg("hideNav", Boolean.TRUE);
+    }
+
+    @Path("latest")
+    public Resource getLatest() {
+        List<DistributionSnapshot> snaps = listPersistedDistributions();
+        Optional<DistributionSnapshot> distribution = snaps.stream()
+                                                           .filter(snap -> snap.getName()
+                                                                               .toLowerCase()
+                                                                               .startsWith("nuxeo platform"))
+                                                           .findFirst();
+
+        String latest = "current";
+        if (distribution.isPresent()) {
+            latest = distribution.get().getKey();
+        }
+        return ctx.newObject("redirectWO", "latest", latest);
+    }
+
+    @Path("{distributionId}")
+    public Resource viewDistribution(@PathParam("distributionId") String distributionId) {
+        if (distributionId == null || "".equals(distributionId)) {
+            return this;
+        }
+
+        List<DistributionSnapshot> snaps = getSnapshotManager().listPersistentSnapshots((ctx.getCoreSession()));
+        if (distributionId.matches(VERSION_REGEX.toString())) {
+            String finalDistributionId = distributionId;
+            String distribution = snaps.stream()
+                                       .filter(s -> s.getVersion().equals(finalDistributionId))
+                                       .findFirst()
+                                       .map(DistributionSnapshot::getKey)
+                                       .orElse("current");
+
+            return ctx.newObject("redirectWO", finalDistributionId, distribution);
+        }
+
+        String orgDistributionId = distributionId;
+        Boolean embeddedMode = Boolean.FALSE;
+        if ("adm".equals(distributionId)) {
+            embeddedMode = Boolean.TRUE;
+        } else {
+            snaps.add(getSnapshotManager().getRuntimeSnapshot());
+            distributionId = SnapshotResolverHelper.findBestMatch(snaps, distributionId);
+        }
+        if (distributionId == null || "".equals(distributionId)) {
```

Comment:
You could use `StringUtils#isBlank`.

Reply:
indeed but i only moved that code -> i would prefer not modifying it to avoid unforeseen regressions in these logics